lib/charms/magma_orchestrator_interface/v0/magma_orchestrator_interface.py

```diff
@@ -238,3 +238,54 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
             fluentd_address=remote_app_relation_data["fluentd_address"],
             fluentd_port=int(remote_app_relation_data["fluentd_port"]),
         )
+
+
+class OrchestratorProvides(Object):
+    """Class to be instantiated by charms providing connectivity with Orchestrator."""
+
+    def __init__(self, charm: CharmBase, relationship_name: str):
+        """Init."""
+        super().__init__(charm, relationship_name)
+        self.relationship_name = relationship_name
+        self.charm = charm
+
+    def set_orchestrator_information(
+        self,
+        root_ca_certificate: str,
+        orchestrator_address: str,
+        bootstrapper_address: str,
+        fluentd_address: str,
+        orchestrator_port: int = 443,
+        bootstrapper_port: int = 443,
+        fluentd_port: int = 24224,
+    ):
+        """Sets orchestrator information in application relation data.
+
+        Args:
+            root_ca_certificate: Orchestrator Root CA Certificate
+            orchestrator_address: Orchestrator address (ex. controller.yourdomain.com)
+            bootstrapper_address: Bootstrapper address (ex. bootstrapper-controller.yourdomain.com)
+            fluentd_address: Fluentd Address (ex. fluentd.yourdomain.com)
+            orchestrator_port: Orchestrator port (Default: 443)
+            bootstrapper_port: Bootstrapper port (Default: 443)
+            fluentd_port: Fluentd port (Default: 24224)
+
+        Returns:
+            None
+        """
+        if not self.charm.unit.is_leader():
+            raise RuntimeError("Unit must be leader to set application relation data.")
+        relation = self.model.get_relation(self.relationship_name)
+        if not relation:
+            raise RuntimeError(f"Relation {self.relationship_name} not yet created")
```

Comment:
I'm not sure what we would return here. If the charm author wrote something that includes writing data to a relation that does not exist it is important that they change their design.

Reply:
in that case they would get an error in both cases, either the library throws an exception or not, no? so it should be their responsibility.

Reply:
If the lib doesn't throw an exception, how would the charm "know" that what it did was bad

Reply:
I would expect the charm to know if a relation is created or not when using it. 
But I understand how this serves as an extra feedback to avoid errors, so it's okay :)